sys/net/gnrc/netapi/gnrc_netapi.c

```diff
@@ -94,52 +94,50 @@ static inline int _snd_rcv_mbox(mbox_t *mbox, uint16_t type, gnrc_pktsnip_t *pkt
 int gnrc_netapi_dispatch(gnrc_nettype_t type, uint32_t demux_ctx,
                          uint16_t cmd, gnrc_pktsnip_t *pkt)
 {
-    int numof = gnrc_netreg_num(type, demux_ctx);
+    int numof = 0;
+    gnrc_netreg_entry_t *sendto = NULL;
```

Comment:
Do a `gnrc_netreg_lookup()` here and a `do { ... } while (gnrc_netreg_getnext());` then you don't need `gnrc_netreg_lookup_next()` here.

Reply:
won't work, if lookup returns `NULL`, so I need to check that *before* entering the loop.